crates/core/src/addon.rs

```diff
@@ -480,14 +480,13 @@ impl Addon {
                 && info.file.modules.iter().any(|m| m.foldername == f.id)
         }) {
             f.id.clone()
+        } else if let Some(f) = addon_folders
+            .iter()
+            .find(|f| info.file.modules.iter().any(|m| m.foldername == f.id))
+        {
+            f.id.clone()
         } else {
-            addon_folders
-                .iter()
-                .find(|f| info.file.modules.iter().any(|m| m.foldername == f.id))
-                .as_ref()
-                .unwrap()
-                .id
-                .clone()
+            info.file.file_name.clone()
```

Comment:
Backlog for you @tarkah.
We had a crash here. Previously you did: 

```rs
.find(|f| info.file.modules.iter().any(|m| m.foldername == f.id)).as_ref().unwrap()
```

Apparently that is not always valid. Who could have known.